back/src/workout/workout.service.spec.ts

```diff
@@ -125,10 +141,20 @@ describe('Workout Service', () => {
 
   it('should soft delete a workout', async () => {
     const [workout] = await workoutRepository.find()
-    let expectedWorkout = new UpdateResult()
+    const expectedWorkout = new UpdateResult()
 
     const softDeletedWorkout = await workoutService.softDelete(workout.id)
 
     expect(softDeletedWorkout).toStrictEqual(expectedWorkout)
   })
+
+  it('should get a workouts by programId', async () => {
+    const programId = HardCodedValuesEnum.programId
+
+    const [expectedWorkout] = await workoutRepository.findByProgramId(programId)
```

Comment:
You want the test to be comprehensive. You need to test every workouts in one. Not only the first one.

Reply:
You should delete the whole 154 line. You already injected an InMemory implementations.

Reply:
We worked together on it, you might want to rework this solution again maybe :

```
  it('should get a workouts by programId', async () => {
    const programId = 'my uuid'
    const [expectedWorkouts] = (await workoutRepository.find()).filter(
      (workout) => workout.program.id === programId,
    )

    const [retrievedWorkout] = await workoutService.getByProgramId('my uuid')

    expect(retrievedWorkout).toStrictEqual(expectedWorkouts)
  })
  ```